Xamarin.Forms.Core/Element.cs

```diff
@@ -208,9 +208,37 @@ public Element Parent
 				OnParentSet();
 
 				OnPropertyChanged();
+
+				if (this.IsTemplateRoot)
+					this.TemplatedParent = value;
 			}
 		}
 
+		internal event EventHandler TemplatedParentChanged;
```

Comment:
why not use PropertyChanged ?
```suggestion
```

Reply:
We could. I thought it would be more efficient this way because it's a highly specific event that is usually only going to be triggered once per `Element`, and will have a relatively small number of listeners compared to `PropertyChanged`.